python/semantic_kernel/agents/chat_history_handler.py

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from abc import ABC, abstractmethod
+from collections.abc import AsyncIterable
+
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.contents.chat_message_content import ChatMessageContent
+from semantic_kernel.contents.streaming_chat_message_content import StreamingChatMessageContent
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+
+@experimental_class
+class ChatHistoryHandler(ABC, KernelBaseModel):
```

Comment:
and this one might not be needed, could be replaced either by nothing or a protocol, let's discuss!

Reply:
The channel concept will come into play soon with other agent types.

Reply:
Also updated to be a protocol, which could serve better here, based on my reading. Thanks.

Reply:
Hmm, mypy isn't happy about a lot of things after changing to Protocol. Will keep as ABC for now, and investigate.